apis/core/v1alpha2/readinessprovider_types.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ReadinessProviderState string
+
+const (
+	ProviderSuccessState    = ReadinessProviderState("success")
+	ProviderFailureState    = ReadinessProviderState("failure")
+	ProviderInProgressState = ReadinessProviderState("inprogress")
+)
+
+type ReadinessConditionState string
+
+const (
+	ConditionSuccessState    = ReadinessConditionState("success")
+	ConditionFailureState    = ReadinessConditionState("failure")
+	ConditionInProgressState = ReadinessConditionState("inprogress")
+)
+
+// ReadinessProviderSpec defines the desired state of ReadinessProvider
+type ReadinessProviderSpec struct {
+	// CheckRef is the name of the check that the current provider satisfies
+	CheckRefs []string `json:"checkRefs"`
+
+	// RepeatInterval is the re-evaluation interval;
+	// if RepeatInterval is not provided or nil,
+	// the provider will be evaluated only once and the evaluation will not be repeated.
+	// A valid value specifies a duration string, such as "1.5h", "1h10m" or "200s".
+	// Refer: https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Duration
+	//+kubebuilder:validation:Optional
+	RepeatInterval *metav1.Duration `json:"repeatInterval"`
```

Comment:
We don't know before hand the execution time of the provider conditions that the consumer is going to provide. The consumer can configure a provider that runs for 30 minutes to provide the result and there is no point in requeuing them after few minutes.

Also, the consumer may choose to run some computation heavy checks like FIPS compliance and CIS benchmarks once in a day to avoid loading clusters. This field helps them in these scenarios.

Reply:
Please ignore the above comment. It makes sense to requeue at a given smaller interval. We can ignore and re-queue if the process is long running!

Reply:
Removed.